modules/game_bot/default_configs/vBot_4.7/vBot/extras.lua

```diff
@@ -0,0 +1,663 @@
+    if not wsadWalking then return end
+    if not settings.holdMwall then return end
+    if m.isOff() then return end
+    if keys ~= mwHot and keys ~= wgHot then return end
+    hold = now
+
+    local tile = getTileUnderCursor()
+    if not tile then return end
+
+    if tile:getText():len() > 0 then
+      tile:setText("")
+    else
+      if keys == mwHot then
+        tile:setText("HOLD MW")
+      else
+        tile:setText("HOLD WG")
+      end
+      table.insert(candidates, tile:getPosition())
+    end
+  end)
+
+  onKeyPress(function(keys)
+    local wsadWalking = modules.game_console.isEnabledWASD()
+    if not wsadWalking then return end
+    if not settings.holdMwall then return end
+    if m.isOff() then return end
+    if keys ~= mwHot and keys ~= wgHot then return end
+
+    if (hold - now) < -1000 then
+      candidates = {}
+      for i, tile in ipairs(g_map.getTiles(posz())) do
+        local text = tile:getText()
+        if text:find("HOLD") then
+          tile:setText("")
+        end
+      end
+    end
+  end)
+end
+
+addCheckBox("checkPlayer", "Check Players", true, rightPanel,
+  "Auto look on players and mark level and vocation on character model")
+if true then
+  local found
+  local function checkPlayers()
+    for i, spec in ipairs(getSpectators()) do
+      if spec:isPlayer() and spec:getText() == "" and spec:getPosition().z == posz() and spec ~= player then
+        g_game.look(spec)
+        found = now
+      end
+    end
+  end
+  if settings.checkPlayer then
+    schedule(500, function()
+      checkPlayers()
+    end)
+  end
+
+  onPlayerPositionChange(function(x, y)
+    if not settings.checkPlayer then return end
+    if x.z ~= y.z then
+      schedule(20, function() checkPlayers() end)
+    end
+  end)
+
+  onCreatureAppear(function(creature)
+    if not settings.checkPlayer then return end
+    if creature:isPlayer() and creature:getText() == "" and creature:getPosition().z == posz() and creature ~= player then
+      g_game.look(creature)
+      found = now
+    end
+  end)
+
+  local regex = [[You see ([^\(]*) \(Level ([0-9]*)\)((?:.)* of the ([\w ]*),|)]]
+  onTextMessage(function(mode, text)
+    if not settings.checkPlayer then return end
+
+    local re = regexMatch(text, regex)
+    if #re ~= 0 then
+      local name = re[1][2]
+      local level = re[1][3]
+      local guild = re[1][5] or ""
+
+      if guild:len() > 10 then
+        guild = guild:sub(1, 10)  -- change to proper (last) values
+        guild = guild .. "..."
+      end
+      local voc
+      if text:lower():find("sorcerer") then
+        voc = "MS"
+      elseif text:lower():find("druid") then
+        voc = "ED"
+      elseif text:lower():find("knight") then
+        voc = "EK"
+      elseif text:lower():find("paladin") then
+        voc = "RP"
+      end
+      local creature = getCreatureByName(name)
+      if creature then
+        creature:setText("\n" .. level .. voc .. "\n" .. guild)
+      end
```

Comment:
```suggestion
      local voc
      if text:lower():find("sorcerer") then
        voc = "MS"
      elseif text:lower():find("druid") then
        voc = "ED"
      elseif text:lower():find("knight") then
        voc = "EK"
      elseif text:lower():find("paladin") then
        voc = "RP"
      end
      if not voc then
        return false
      end
      local creature = getCreatureByName(name)
      if creature then
        creature:setText("\n" .. level .. voc .. "\n" .. guild)
      end
```

Reply:
if not voc then
        return false
      end

fix this in custom server or old protocol

`ERROR: [BOT] [string "/vBot/extras.lua"]:608: attempt to concatenate local 'voc' (a nil value)`